### ISSUE TITLE
[3.0] Save icon missing

### ISSUE THREAD

**Issue by slavarazum** [froala/wysiwyg-editor#3525]
Icon for `save` button is missing.

![screenshot](https://image.prntscr.com/image/1iW0K5flRpWrBl7o9uDcjw.png)

SVG code from inspector:

```html
<svg class="fr-svg" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
    <path d="undefined"></path>
</svg>
```

##### Steps to reproduce the problem.
Just try to enable `save` button

##### Editor version.
3.0.1

##### OS.
Mac OS Mojave (I think It does not affect)

##### Browser.
Any (I think It does not affect)

**Comment by casualuser** [comment]
Hello,

It looks like FA was not enabled /js/plugins/font_awesome.min.js so Save icon missed.
Please try to add FA plugin and verify. If issue somehow persists then to reproduce this with JSFiddle

**Comment by slavarazum** [comment]
What if I don't want to use Font Awesome? I want to use base icons.

**Comment by casualuser** [comment]
ok, we need to recheck this for base icons
can you please provide jsFiddle with your configuration?

**Comment by slavarazum** [comment]
https://jsfiddle.net/30asLepq/1/

`Save` button appears after `Bold`

**Comment by casualuser** [comment]
thx, reporting this to developers
will keep you updated with progress

**Comment by slavarazum** [comment]
Thanks! But this issue https://github.com/froala/wysiwyg-editor/issues/3526 is more important.

**Comment by casualuser** [comment]
processing #3526 now

and here something from our developers:
> we don't have inbuilt save icon as this is not a feature of froala. However you can still define custom svg icons like below

FroalaEditor.DefineIcon('save',{PATH:'M3,17v2h6v-2H3z M3,5v2h10V5H3z M13,21v-2h8v-2h-8v-2h-2v6H13z M7,9v2H3v2h4v2h2V9H7z M21,13v-2H11v2H21z M15,9h2V7h4V5h-4 V3h-2V9z'})

Here is a sample https://jsfiddle.net/froala/mjvwrg9q/

**Comment by slavarazum** [comment]
@casualuser So, you have save button with specific functionality but don't support an icon for it? Sounds strange...

**Comment by slavarazum** [comment]
@casualuser
How do you think, is it appropriate behavior, when you have save button, but don't have icon for it?
I think it's a bug, totally.

**Comment by frogga67** [comment]
I agree, the save button still works but with no icon, I have tested with fontAwesome included and still there is no icon, using version 3.0.5

![image](https://user-images.githubusercontent.com/7875645/63565818-e98f2f80-c5ad-11e9-86bc-b03103c53b9a.png)

**Comment by jwhitmarsh** [comment]
> processing #3526 now
> 
> and here is something from our developers:
> 
> > we don't have inbuilt save icon as this is not a feature of froala. However you can still define custom svg icons like below
> 
> FroalaEditor.DefineIcon('save',{PATH:'M3,17v2h6v-2H3z M3,5v2h10V5H3z M13,21v-2h8v-2h-8v-2h-2v6H13z M7,9v2H3v2h4v2h2V9H7z M21,13v-2H11v2H21z M15,9h2V7h4V5h-4 V3h-2V9z'})
> 
> Here is a sample https://jsfiddle.net/froala/mjvwrg9q/

For me this comes out as one of FontAwesome's settings icons:

![image](https://user-images.githubusercontent.com/8026009/64188959-525a8f80-ce6b-11e9-92f5-54de34d088e9.png)

I don't understand why FA icons aren't working in V3 any more? This configuration below used to work with no fuss:
```js
window.FroalaEditor.DefineIcon("addReference", {
  NAME: "bookmark"
});
```

**Comment by jwhitmarsh** [comment]
For anyone coming to this in the future, the fix was to specify a template. Important to know that the `template` property is a key of the `ICON_TEMPLATES` dict/enum, not an actual template. 

To explain, this doesn't work:
```js
FroalaEditor.DefineIcon("customSave", {
  NAME: "floppy-o",
  template: "<i class="fa fa-[NAME]" aria-hidden="true"></i>"
});
```

Nor does this:
```js
FroalaEditor.DefineIcon("customSave", {
  NAME: "floppy-o",
  template: FroalaEditor.ICON_TEMPLATES.font_awesome
});
```

This does:
```js
FroalaEditor.DefineIcon("customSave", {
  NAME: "floppy-o",
  template: "font_awesome"
});
```

**Comment by frogga67** [comment]
Just upgraded to 3.0.6, icon still missing, would really like to see this fixed.

**Comment by Lee1984** [comment]
Similar to @jwhitmarsh although not the save button like in https://github.com/froala/wysiwyg-editor/issues/3470

Similar problem narrowed down to.

Upgrade from v 2x to 3.0.6

    FroalaEditor.DefineIcon('atIcon', {NAME: 'at'});

I do not know how font awesome works but in v 2x for the `at` icon it renders

    <i class="fa fa-at" aria-hidden="true"></i>

and the icon appears. However in v 3.0.6 you get a svg file output

    <svg class="fr-svg" focusable="false" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d=""></path></svg>

Change code and define the template and set it to `fas`

      FroalaEditor.DefineIconTemplate('font_awesome_5s', '<i class="fas fa-[FA5NAME]" aria-hidden="true"></i>');
      FroalaEditor.DefineIcon('atIcon', {FA5NAME: 'at', template: 'font_awesome_5s'});

Output is `far`, eg

    <i class="far fa-at" aria-hidden="true"></i>

**Comment by frogga67** [comment]
Still not fixed with 3.1.0

@jwhitmarsh solution worked with a small tweak:

`FroalaEditor.DefineIcon("save", { NAME: "floppy-o fa-lg", template: "font_awesome"});`
